ipa-core/src/protocol/ipa_prf/prf_sharding/feature_label_dot_product.rs

```diff
@@ -94,19 +101,23 @@ impl InputsRequiredFromPrevRow {
         )
         .await?;
 
-        let capped_attributed_feature_vector = select(
-            ctx.narrow(&Step::ComputedCappedFeatureVector),
+        let condition: Replicated<Boolean, B> = Replicated::<Boolean, B>::expand(&capped_label);
+        let mut bit_decomposed_output = BitDecomposed::new(iter::empty());
+        bit_decomposed_output
+            .transpose_from(&input_row.feature_vector)
+            .unwrap_infallible();
+        let capped_attributed_feature_vector = bool_and(
+            ctx,
             record_id,
-            &capped_label,
-            &input_row.feature_vector,
-            &Replicated::<FV>::ZERO,
+            &bit_decomposed_output,
+            repeat_n(&condition, input_row.feature_vector.len()),
```

Comment:
Whoops... This used to be right, but then I decided to move the `transpose` call up sooner, so that when I called `bool_and` the internal steps that use `BitOpStep` would *actually* be per bit, and not abuse that name to actually be per-bucket. I didn't update this after I made that change.